### PR TITLE
PR: Fix issue where Spyder's inline graphics preferences were not applied

### LINKED ISSUE #10932: Spyder is ignoring matplotlib user settings
### Issue Report Checklist

* [X] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [X] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [X] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [X] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [X] Tried basic troubleshooting (if a bug/error)
    * [X] Restarted Spyder
    * [X] Reset preferences with ``spyder --reset``
    * [X] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [X] Tried the other applicable steps from the Troubleshooting Guide
* [X] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
If I uncheck "Activate support" under Tools > Preferences > IPython > Graphics, my plots are still going to the plot pane, and not to a separate plot window. In previous versions, I always had the graphics support disabled and handled all matplotlib stuff externally, and it always worked perfectly. Now, even with the environmental variable "MPLBACKEND" set to "TkCairo" and the backend set in matplotlibrc, I still get this:

```python
In [1]: import matplotlib.pyplot as plt

In [2]: plt.get_backend()
Out[2]: 'module://ipykernel.pylab.backend_inline'
```

If I instead import it like this:
```python
In [1]: from matplotlib import rcParams

In [2]: rcParams.update({'backend': 'TkCairo'})

In [3]: from matplotlib import pyplot as plt

In [4]: plt.get_backend()
Out[4]: 'TkCairo'
```
everything seems fine, but plots are still shown as non-vectorial images in the plot pane

```python

In [5]: plt.plot([1,2,3,4,3,2,1])
Out[5]: [<matplotlib.lines.Line2D at 0x7f21f36ff750>]

Figures now render in the Plots pane by default. To make them also appear inline in the Console, uncheck "Mute Inline Plotting" under the Plots pane options menu. 

```

If I activate IPython graphics support and select Tkinter as backend, it works as expected, except that it do not wait for the `plt.show()` call.

### What steps reproduce the problem?

1. Set matplotlib backend as anything but inline in MPLBACKEND and/or matplotlibrc
2. Disable IPython graphics support
3. `import matplotlib.pyplot as plt`
4. `plt.plot([1,2,3])`

### What is the expected output? What do you see instead?
Expected: no output until `plt.show()` is called, and when it is, a separate window pop out showing the vectorial plot.
Observed: rendered non-vectorial plot is immediately shown in the plot pane.

## Versions

* Spyder version: Spyder 4.0.0rc3 
* Python version: Python 3.7.5 64-bit
* Qt version: Qt 5.9.6 
* PyQt version: PyQt5 5.9.2
* Operating System name/version: Gentoo GNU/Linux

### Dependencies

```
cloudpickle >=0.5.0          :  1.2.2 (OK)
pygments >=2.0               :  2.4.2 (OK)
qtconsole >=4.6.0            :  4.6.0 (OK)
nbconvert >=4.0              :  5.6.1 (OK)
sphinx >=0.6.6               :  2.2.1 (OK)
pylint >=0.25                :  2.4.4 (OK)
psutil >=0.3                 :  5.6.5 (OK)
qtawesome >=0.5.7            :  0.6.0 (OK)
qtpy >=1.5.0                 :  1.9.0 (OK)
pickleshare >=0.4            :  0.7.5 (OK)
zmq >=17                     :  18.1.0 (OK)
chardet >=2.0.0              :  3.0.4 (OK)
numpydoc >=0.6.0             :  0.9.1 (OK)
spyder_kernels >=1.8.0;<2.0.0:  1.8.0 (OK)
qdarkstyle >=2.7             :  2.7 (OK)
atomicwrites >=1.2.0         :  1.3.0 (OK)
diff_match_patch >=20181111  :  20181111 (OK)
intervaltree                 :  None (OK)
watchdog                     :  None (OK)
keyring                      :  None (OK)
pexpect >=4.4.0              :  4.7.0 (OK)
pympler                      :  None (OK)
sympy >=0.7.3                :  None (NOK)
cython >=0.21                :  None (NOK)
IPython >=4.0                :  7.9.0 (OK)
matplotlib >=2.0.0           :  3.1.2 (OK)
pandas >=0.13.1              :  None (NOK)
numpy >=1.7                  :  1.17.4 (OK)
scipy >=0.17.0               :  1.3.3 (OK)
pyls >=0.31.1;<0.32.0        :  0.31.1 (OK)
xdg >=0.26                   :  0.26 (OK)
rtree >=0.8.3                :  0.8.3 (OK)
```

Btw, I loved the new features and visual of those 4.0 releases, thanks a lot for all the work in this amazing IDE :)

### LINKED ISSUE #22034: IPython console Matplotlib rc parameters are inconsistent when switching backends
## Problem Description

Spyder's IPython Console does not behave consistently with respect to Matplotlib rc settings when switching backends. The following table summarizes the problem.

| Backend Preference | Toggle Backend | Expected Params | Actual Params |
|--------|--------|--------|--------|
| Inline | initial | Spyder inline preferences | Spyder inline preferences |
| Inline | to interactive | matplotlibrc | Spyder kernel default |
| Inline | back to inline | Spyder inline preferences | Spyder kernel default |
| Interactive | initial | matplotlibrc | Spyder inline preferences |
| Interactive | to inline | Spyder inline preferences | Spyder kernel default |
| Interactive | back to interactive | matplotlibrc | Spyder inline preferences |

With inline graphics backend:
```python
In [1]: import matplotlib as mpl

In [2]: mpl.rcParams['savefig.format'], mpl.rcParams['figure.dpi'], mpl.rcParams['figure.figsize'], mpl.rcParams['font.size'], mpl.rcParams['figure.subplot.bottom']
Out[2]: ('pdf', 144.0, [6.0, 4.0], 8.0, 0.05)

In [3]: %matplotlib auto
Using matplotlib backend: QtAgg
Using matplotlib backend: QtAgg

In [4]: mpl.rcParams['savefig.format'], mpl.rcParams['figure.dpi'], mpl.rcParams['figure.figsize'], mpl.rcParams['font.size'], mpl.rcParams['figure.subplot.bottom']
Out[4]: ('pdf', 72.0, [6.0, 4.0], 10.0, 0.125)

In [5]: %matplotlib inline

In [6]: mpl.rcParams['savefig.format'], mpl.rcParams['figure.dpi'], mpl.rcParams['figure.figsize'], mpl.rcParams['font.size'], mpl.rcParams['figure.subplot.bottom']
Out[6]: ('pdf', 72.0, [6.0, 4.0], 10.0, 0.125)

In [7]: mpl.rc_file_defaults()

In [8]: mpl.rcParams['savefig.format'], mpl.rcParams['figure.dpi'], mpl.rcParams['figure.figsize'], mpl.rcParams['font.size'], mpl.rcParams['figure.subplot.bottom']
Out[8]: ('pdf', 99.0, [7.0, 5.0], 9.0, 0.1)
```
As shown, after switching to Qt backend, the parameters are updated to the kernel defaults (https://github.com/spyder-ide/spyder-kernels/blob/09c370aab411c954f869649279481636f666098b/spyder_kernels/console/start.py#L101C1-L116C1). Except that the `savefig.format`

## Versions

* Spyder version: 6.0.0a5  (standalone)
* Python version: 3.10.13 64-bit
* Qt version: 5.15.8
* PyQt5 version: 5.15.9
* Operating System: macOS-14.4.1-x86_64-i386-64bit

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0              :  1.4.1 (OK)
chardet >=2.0.0                   :  5.2.0 (OK)
cloudpickle >=0.5.0               :  3.0.0 (OK)
cookiecutter >=1.6.0              :  2.5.0 (OK)
diff_match_patch >=20181111       :  20230430 (OK)
intervaltree >=3.0.2              :  3.1.0 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1  :  8.21.0 (OK)
jedi >=0.17.2,<0.20.0             :  0.19.1 (OK)
jellyfish >=0.7                   :  1.0.3 (OK)
jsonschema >=3.2.0                :  4.21.1 (OK)
keyring >=17.0.0                  :  24.3.0 (OK)
nbconvert >=4.0                   :  7.16.0 (OK)
numpydoc >=0.6.0                  :  1.6.0 (OK)
parso >=0.7.0,<0.9.0              :  0.8.3 (OK)
pexpect >=4.4.0                   :  4.9.0 (OK)
pickleshare >=0.4                 :  0.7.5 (OK)
psutil >=5.3                      :  5.9.8 (OK)
pygments >=2.0                    :  2.17.2 (OK)
pylint >=3.1,<4                   :  3.1.0 (OK)
pylint_venv >=3.0.2               :  3.0.3 (OK)
pyls_spyder >=0.4.0               :  0.4.0 (OK)
pylsp >=1.11.0,<1.12.0            :  1.11.0 (OK)
pylsp_black >=2.0.0,<3.0.0        :  2.0.0 (OK)
pyuca >=1.2                       :  1.2 (OK)
qdarkstyle >=3.2.0,<3.3.0         :  3.2.3 (OK)
qstylizer >=0.2.2                 :  0.2.2 (OK)
qtawesome >=1.3.1,<1.4.0          :  1.3.1 (OK)
qtconsole >=5.5.1,<5.6.0          :  5.5.1 (OK)
qtpy >=2.4.0                      :  2.4.1 (OK)
rtree >=0.9.7                     :  1.2.0 (OK)
setuptools >=49.6.0               :  69.0.3 (OK)
sphinx >=0.6.6                    :  7.2.6 (OK)
spyder_kernels >=3.0.0b5,<3.0.0b6 :  3.0.0b5 (OK)
superqt >=0.6.1,<1.0.0            :  0.6.1 (OK)
textdistance >=4.2.0              :  4.5.0 (OK)
three_merge >=0.1.1               :  0.1.1 (OK)
watchdog >=0.10.3                 :  4.0.0 (OK)
zmq >=24.0.0                      :  25.1.2 (OK)

# Optional:
cython >=0.21                     :  3.0.8 (OK)
matplotlib >=3.0.0                :  3.8.2 (OK)
numpy >=1.7                       :  1.26.4 (OK)
pandas >=1.1.1                    :  2.2.0 (OK)
scipy >=0.17.0                    :  1.12.0 (OK)
sympy >=0.7.3                     :  1.12 (OK)
```

### PR DESCRIPTION
**Pull request by mrclary**
## Description of Changes

- Depends on https://github.com/spyder-ide/spyder-kernels/pull/487.

### Issue(s) Resolved

Fixes #22034
Fixes #10932

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

**Review by ccordoba12 — COMMENTED** [review]
This looks good to me but I think it's missing a test to check that Matplotlib rcParams are effectively different between the inline and interactive backends.

**Comment by mrclary** [comment]
> This looks good to me but I think it's missing a test to check that Matplotlib rcParams are effectively different between the inline and interactive backends.

Good idea. I'll create a test.

**Review by ccordoba12 — COMMENTED** [review]
Thanks @mrclary for the new test!

File: spyder/plugins/ipythonconsole/tests/conftest.py [hunk 1]
```diff
@@ -342,3 +340,23 @@ def threads_condition():
         files = [repr(f) for f in proc.open_files()]
         show_diff(init_files, files, "files")
         raise
+
+
+@pytest.fixture
+def mpl_rc_file():
+    """Create matplotlibrc file"""
+    file_contents = """\
```
[the remark this answers is not in the document]
Comment (mrclary) [reply spyder/plugins/ipythonconsole/tests/conftest.py:348]: No, it is not necessary. Without it, there is a blank line at the top of the file; but this has no meaningful consequence.

File: spyder/plugins/ipythonconsole/tests/conftest.py [hunk 2]
```diff
@@ -342,3 +340,23 @@ def threads_condition():
         files = [repr(f) for f in proc.open_files()]
         show_diff(init_files, files, "files")
         raise
+
+
+@pytest.fixture
+def mpl_rc_file():
```
[the remark this answers is not in the document]
Comment (mrclary) [reply spyder/plugins/ipythonconsole/tests/conftest.py:346]: Okay. `TEMP_DIRECTORY` appears to be used only for `test_console_disambiguation`. Should that test also use the `tmp_path` fixture instead, and remove the `TEMP_DIRECTORY` constant?

File: spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py [hunk 3]
```diff
@@ -1983,6 +1995,82 @@ def test_mpl_conf(ipyconsole, qtbot):
     mock.assert_called_once_with({'pylab/inline/bottom': 0.314})
 
 
+def test_rc_params(mpl_rc_file, ipyconsole, qtbot):
+    """
+    Test that rcParams are correctly set/reset when changing backends and
+    setting inline preferences.
+    """
+    # import pdb; pdb.set_trace()
```
[the remark this answers is not in the document]
Comment (mrclary) [reply spyder/plugins/ipythonconsole/tests/test_ipythonconsole.py:2003]: Oops. Thanks.

File: spyder/plugins/ipythonconsole/tests/conftest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ccordoba12) [reply spyder/plugins/ipythonconsole/tests/conftest.py]: Sure, if that's ok for you, please do it here.

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @mrclary!

**Pull request merged**